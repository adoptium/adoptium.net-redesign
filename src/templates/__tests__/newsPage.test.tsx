import React from "react"
import { render } from "@testing-library/react"
import NewsPage, { Head } from "../newsPage"
import { describe, expect, it } from "vitest"
import { axe } from "vitest-axe"
import { createMDXData } from "../../__fixtures__/page"

let mockData = createMDXData()
const pageContext = {
  currentPage: 2,
  previousPageNumber: 1,
  nextPageNumber: 3,
  totalPages: 5, 
}

describe("News Template page", () => {
  it("renders correctly", () => {
    const { container } = render(
      <NewsPage data={mockData} pageContext={pageContext} />,
    )
    // eslint-disable-next-line
    const pageContent = container.querySelector("main")

    expect(pageContent).toMatchSnapshot()
  })

  it("head renders correctly", () => {
    const { container } = render(<Head pageContext={pageContext} />)
    // eslint-disable-next-line
    const title = container.querySelector("title")
    expect(title?.textContent).toEqual("News & Events â€“ Page 2 | Adoptium")
  })

  it("has no accessibility violations", async () => {
    const { container } = render(
      <NewsPage data={mockData} pageContext={pageContext} />,
    )
    const results = await axe(container)
    expect(results).toHaveNoViolations()
  })
})
