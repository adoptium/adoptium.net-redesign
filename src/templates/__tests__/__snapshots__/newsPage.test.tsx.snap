// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`News Template page > renders correctly 1`] = `
<main>
  <div
    class="relative pt-40 pb-16 md:pb-36 md:pt-60 overflow-hidden"
  >
    <div
      class="w-full relative"
    >
      <div
        class="absolute z-[-1] w-[2396px] h-[1340px] left-[23px] md:left-[112px] top-[21px] md:top-[171px] bg-[#410170] shadow-[0_0_400px_rgba(65,1,112,1)] rounded-full blur-[400px]"
      />
      <div
        class="absolute z-[-1] w-[1487px] h-[893px] left-[120px] md:left-[676px] top-[120px] md:top-[395px] bg-[#B62987] shadow-[0_0_400px_rgba(182,41,135,1)] rounded-full blur-[400px]"
      />
      <div
        class="absolute z-[-1] w-[688px] h-[446px] left-[400px] md:left-[1131px] top-[618px] bg-[#FE8492] shadow-[0_0_400px_rgba(254,132,146,1)] rounded-full blur-[400px]"
      />
    </div>
    <div
      class="mx-auto max-w-[1048px] w-full px-6 lg:px-0 flex flex-col items-center justify-center"
    >
      <div
        class="self-stretch flex-col justify-center items-center gap-6 flex"
      >
        <div
          class="self-stretch  flex-col justify-center items-center gap-4 flex"
        >
          <div
            class="justify-start items-center gap-3 inline-flex"
          >
            <svg
              fill="none"
              height="14"
              viewBox="0 0 14 14"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="#FF1464"
                height="8"
                rx="2"
                width="8"
                x="3"
                y="3"
              />
              <rect
                height="11"
                rx="3.5"
                stroke="#FF1464"
                stroke-opacity="0.25"
                stroke-width="3"
                width="11"
                x="1.5"
                y="1.5"
              />
            </svg>
            <div
              class="text-rose-600 text-base font-bold leading-normal"
            >
              News
            </div>
          </div>
          <div
            class="self-stretch text-center text-white   text-[56px] lg:text-[80px] leading-[114.286%] md:leading-[120%] font-semibold"
          >
            News & Updates
          </div>
        </div>
        <div
          class="self-stretch text-center text-lightgrey text-[20px] font-normal leading-[140%] mx-auto max-w-[860px] px-2 w-full"
        >
          Follow the latest updates from the Eclipse Adoptium Project
        </div>
      </div>
    </div>
  </div>
  <div
    class="max-w-[1264px] mx-auto px-6 py-8 md:pt-12"
  >
    <div
      class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6 justify-items-center py-8 md:py-12"
    >
      <div
        class="flex flex-col max-w-[385px] w-full sm:w-[385px] h-[550px] rounded-3xl newscard-2 bg-[#200D46] animate-pulse"
      >
        <div
          class="m-0.5 rounded-t-3xl h-[200px] bg-gradient-to-r from-purple-900 to-purple-800 w-full"
        />
        <div
          class="p-8 flex flex-col h-[350px]"
        >
          <div
            class="h-[70px]"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[24px] w-3/4 rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[24px] w-1/2 rounded"
            />
          </div>
          <div
            class="flex justify-between py-3"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[20px] w-[100px] rounded"
            />
          </div>
          <div
            class="flex-grow overflow-hidden h-[100px]"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-full rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-full rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-3/4 rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-1/2 rounded"
            />
          </div>
          <div
            class="mt-auto"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[40px] w-[120px] rounded-full"
            />
          </div>
        </div>
      </div>
      <div
        class="flex flex-col max-w-[385px] w-full sm:w-[385px] h-[550px] rounded-3xl newscard-2 bg-[#200D46] animate-pulse"
      >
        <div
          class="m-0.5 rounded-t-3xl h-[200px] bg-gradient-to-r from-purple-900 to-purple-800 w-full"
        />
        <div
          class="p-8 flex flex-col h-[350px]"
        >
          <div
            class="h-[70px]"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[24px] w-3/4 rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[24px] w-1/2 rounded"
            />
          </div>
          <div
            class="flex justify-between py-3"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[20px] w-[100px] rounded"
            />
          </div>
          <div
            class="flex-grow overflow-hidden h-[100px]"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-full rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-full rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-3/4 rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-1/2 rounded"
            />
          </div>
          <div
            class="mt-auto"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[40px] w-[120px] rounded-full"
            />
          </div>
        </div>
      </div>
      <div
        class="flex flex-col max-w-[385px] w-full sm:w-[385px] h-[550px] rounded-3xl newscard-2 bg-[#200D46] animate-pulse"
      >
        <div
          class="m-0.5 rounded-t-3xl h-[200px] bg-gradient-to-r from-purple-900 to-purple-800 w-full"
        />
        <div
          class="p-8 flex flex-col h-[350px]"
        >
          <div
            class="h-[70px]"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[24px] w-3/4 rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[24px] w-1/2 rounded"
            />
          </div>
          <div
            class="flex justify-between py-3"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[20px] w-[100px] rounded"
            />
          </div>
          <div
            class="flex-grow overflow-hidden h-[100px]"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-full rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-full rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-3/4 rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-1/2 rounded"
            />
          </div>
          <div
            class="mt-auto"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[40px] w-[120px] rounded-full"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6 justify-items-center py-8 md:py-12"
    >
      <div
        class="flex flex-col max-w-[385px] w-full sm:w-[385px] h-[550px] rounded-3xl newscard-2 bg-[#200D46] animate-pulse"
      >
        <div
          class="m-0.5 rounded-t-3xl h-[200px] bg-gradient-to-r from-purple-900 to-purple-800 w-full"
        />
        <div
          class="p-8 flex flex-col h-[350px]"
        >
          <div
            class="h-[70px]"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[24px] w-3/4 rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[24px] w-1/2 rounded"
            />
          </div>
          <div
            class="flex justify-between py-3"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[20px] w-[100px] rounded"
            />
          </div>
          <div
            class="flex-grow overflow-hidden h-[100px]"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-full rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-full rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-3/4 rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-1/2 rounded"
            />
          </div>
          <div
            class="mt-auto"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[40px] w-[120px] rounded-full"
            />
          </div>
        </div>
      </div>
      <div
        class="flex flex-col max-w-[385px] w-full sm:w-[385px] h-[550px] rounded-3xl newscard-2 bg-[#200D46] animate-pulse"
      >
        <div
          class="m-0.5 rounded-t-3xl h-[200px] bg-gradient-to-r from-purple-900 to-purple-800 w-full"
        />
        <div
          class="p-8 flex flex-col h-[350px]"
        >
          <div
            class="h-[70px]"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[24px] w-3/4 rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[24px] w-1/2 rounded"
            />
          </div>
          <div
            class="flex justify-between py-3"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[20px] w-[100px] rounded"
            />
          </div>
          <div
            class="flex-grow overflow-hidden h-[100px]"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-full rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-full rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-3/4 rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-1/2 rounded"
            />
          </div>
          <div
            class="mt-auto"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[40px] w-[120px] rounded-full"
            />
          </div>
        </div>
      </div>
      <div
        class="flex flex-col max-w-[385px] w-full sm:w-[385px] h-[550px] rounded-3xl newscard-2 bg-[#200D46] animate-pulse"
      >
        <div
          class="m-0.5 rounded-t-3xl h-[200px] bg-gradient-to-r from-purple-900 to-purple-800 w-full"
        />
        <div
          class="p-8 flex flex-col h-[350px]"
        >
          <div
            class="h-[70px]"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[24px] w-3/4 rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[24px] w-1/2 rounded"
            />
          </div>
          <div
            class="flex justify-between py-3"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[20px] w-[100px] rounded"
            />
          </div>
          <div
            class="flex-grow overflow-hidden h-[100px]"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-full rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-full rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-3/4 rounded mb-2"
            />
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[16px] w-1/2 rounded"
            />
          </div>
          <div
            class="mt-auto"
          >
            <div
              class="bg-gradient-to-r from-purple-800 to-purple-700 h-[40px] w-[120px] rounded-full"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex justify-center items-center gap-8 md:gap-14 my-8"
    >
      <div
        class="flex items-center gap-5"
      >
        <div
          class="flex items-center justify-between gap-6 border border-[#3E3355] rounded-[80px] px-6 py-3"
        >
          <div
            class="bg-gradient-to-r from-purple-800 to-purple-700 h-[24px] w-[120px] rounded"
          />
        </div>
      </div>
    </div>
    <div
      class="flex justify-center items-center gap-5 mt-8"
    >
      <div
        class="bg-gradient-to-r from-purple-800 to-purple-700 h-[46px] w-[120px] rounded-full"
      />
    </div>
  </div>
</main>
`;
